File: php/features.php
<?php
declare(strict_types=1);

// Gemini SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class GeminiFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new GeminiBaseFeature();
            case "test":
                return new GeminiTestFeature();
            default:
                return new GeminiBaseFeature();
        }
    }
}

File: py/features.py
# Gemini SDK feature factory

from feature.base_feature import GeminiBaseFeature
from feature.test_feature import GeminiTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GeminiBaseFeature(),
        "test": lambda: GeminiTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
